src/attendees/attendee.py

```diff
@@ -60,26 +100,114 @@ def has_attribute(self, key: Hashable) -> bool:
         Returns:
             bool: True if Attendee has attribute and vice-versa.
         """
-        raise NotImplementedError
-
+    @property
+    def has_fname(self):    
+        return self._fname == None
+    
+    @property
+    def has_lname(self):    
+        return self._lname == None # already gives a boolean, if statements always accept boolean
+
+    # if condtion, if not condition (i.e. else)
+    
+    @property
+    def has_email(self):    
+        return self._email == None
+
+    @property
+    def has_file_url(self):    
+        return self._file_url == None
+
+    @property
+    def has_file_path(self):    
+        return self._file_url == None
+    
     def set_attribute(self, key: Hashable, value: Any,
-                     verifyFunc: VerifyFunc = None):
+                     verifyFunc: VerifyFunc = None): #verifyFunc is a function, a callback - the what, for miscellaneous, 
+                    # i wont write callback, will use it
         """
         Set an arbitrary attribute of the attendee.
 
-        Will add the attribute if it doesn't exist.
+        Will add the attribute if it doesn't exist. First Checks if `verifyFunc`
+        is `None`. If `None`, do not verify value, any value is validOnce verified,
+        set misc attribute using `key`.
 
         Args:
             key (Hashable): The name of the attribute.
             value (Any): The attribute value.
             verifyFunc (Callable[[Hashable, Any], bool]): A function to verify
-            the value being set, if let as None, will not verify value.
+            the value being set, if let as `None`, will not verify value. If
+            `True`, value is valid and vice-versa.
```

Comment:
Small typo in the docstring: it should say "if left as None" instead of "if let as None"